final/src/Enemy.cpp

```diff
@@ -0,0 +1,135 @@
+#include "Enemy.h"
+
+Enemy::Enemy(Player* player, sf::Vector2f spawnPosition)
+        : m_player(player), m_position(spawnPosition), m_isAttacking(false), m_isDestroyed(false), m_points(5) {}
+
+sf::Vector2f Enemy::getPosition() const {
+    return m_position;
+}
+
+sf::Sprite Enemy::getSprite() const {
+    return m_sprite;
+}
+
+int Enemy::getPoints() const {
+    return m_points;
+}
+
+Item* Enemy::getDropItem() const {
+    return m_dropItem;
+};
+
+bool Enemy::isDestroyed() const {
+    return m_isDestroyed;
+};
+
+void Enemy::takeDamage(int damage) {
+    m_health -= damage;
+    if (m_health <= 0) {
+        m_isDestroyed = true;
+    }
+}
+
+void Enemy::moveTowardsPlayer(float deltaTime, float distance, sf::Vector2f& direction) {
+    direction.x /= distance;
+    direction.y /= distance;
+
+    m_position += direction * deltaTime * m_speed;
+    m_sprite.setPosition(m_position);
+    m_sprite.setTexture(m_textureRun);
+}
+
+bool Enemy::checkAttackCooldown() {
+    if (m_attackCooldown.getElapsedTime().asSeconds() >= m_attackInterval) {
+        m_isAttacking = true;
+        m_attackCooldown.restart();
+        return true;
+    }
+    return false;
+}
+
+void Enemy::attackPlayer() {
+    if (m_isAttacking) {
+        m_player->takeDamage(m_damage);
+        m_isAttacking = false;
+        m_sprite.setTexture(m_textureAttack);
+    }
+}
+
+void Enemy::updateAnimation(float deltaTime, const sf::Vector2f& direction) {
+    bool isMovingRight = direction.x < 0;
+    m_sprite.setTextureRect(m_animationSprite->getRect(deltaTime, isMovingRight));
+}
+
+void Enemy::update(float deltaTime) {
+    sf::Vector2f direction = m_player->getPosition() - m_position;
+    float distance = std::sqrt(direction.x * direction.x + direction.y * direction.y);
+
+    if (distance > m_attackDistance) {
+        moveTowardsPlayer(deltaTime, distance, direction);
+    } else {
+        if (checkAttackCooldown()) {
+            attackPlayer();
+        }
+    }
+
+    updateAnimation(deltaTime, direction);
+}
+
+
+
+void Enemy::draw(sf::RenderWindow& window) {
+    window.draw(m_sprite);
+}
+
+Hyena::Hyena(Player* player, sf::Vector2f spawnPosition)
+        : Enemy(player, spawnPosition) {
+    m_speed = 20.f;
+    m_health = 10;
+    m_damage = 10;
+    m_attackInterval = 1.f;
+    m_attackDistance = 40.f;
```

Comment:
It would be great to move out all these constants into one place, some sort of Config header, so you could tweak your gameplay from one place.
In future, you could even move these values into disk file to change your logic without need to recompile the code every time